### PR TITLE
Scale contrastive loss by world size

### PR DESCRIPTION
**Pull request by NohTow**
With the recent addition of gathering from other gpus in a multi-gpu setting, there's a scaling by the world size missing to the loss (because PyTorch does an average over the number of GPUs by default, so we need to scale by the same number to have a similar loss no matter the number of GPUs).
Also took the opportunity to create a dedicated function for getting the rank instead of duplicating code.

I wonder if we should set the world size/rank as a parameter to only read it once instead of at each step... Maybe it's too much optimization idk

### CODE REVIEW

File: pylate/utils/distributed.py [hunk 1]
```diff
@@ -102,3 +102,21 @@ def all_gather_with_gradients(tensor: torch.Tensor) -> Sequence[torch.Tensor]:
...
 
     return [tensor]
+
+
+def get_rank() -> int:
+    """Returns the current rank in a distributed training."""
+    # Check if torch.distributed is properly available and initialized.
+    if dist.is_available() and dist.is_initialized():
+        return dist.get_rank()
+    else:
```
Comment (raphaelsty) [inline pylate/utils/distributed.py:112]: remove the else and simply return 0

**Comment by NohTow** [comment]
@raphaelsty I removed the else and directly return even though I find the explicit way more readable, but it's a matter of test IG
I suppose we can merge after the tests run

**Comment by raphaelsty** [comment]
@NohTow LGTM, once test pass we can merge 🚀
